src/dune_rules/module.mli

```diff
@@ -68,6 +68,8 @@ val has : t -> ml_kind:Ml_kind.t -> bool
 (** Prefix the object name with the library name. *)
 val with_wrapper : t -> main_module_name:Module_name.t -> t
 
+val add_file : t -> Ml_kind.t -> File.t -> t
```

Comment:
Fairly arbitrary interface choice here as it ended up being concise at the call-site; alternatives welcome.